luavm/src/lib/instructions/functions.rs

```diff
@@ -25,11 +31,33 @@ pub fn call(vm: &mut Vm, instr: u32) -> Result<(), LuaError> {
     for i in 1..reg_num {
         vm.stack.push(vm.registers[i].clone());
     }
+    // the compiler might have pushed some arguments, but the exact number is encoded
+    // in the second operand of the call instruction
+    // we have to make sure that those arguments are copied where the function expects
+    // its parameters to be located at
+    let num_of_args = second_arg(instr) as usize;
+    let mut index_of_arg = vm.stack.len() - (reg_num - 1) - num_of_args;
+    let num_of_params = vm.bytecode.get_function(vm.curr_func).param_count();
+    // copy arguments into registers [R(1)..R(num_of_params)]
+    for i in 0..num_of_params {
+        // if the caller didn't push enough arguments, we have to set the remaining
```

Comment:
Is this Lua's semantics? You can call a function with fewer arguments than it requires?

Reply:
Yes to both questions.
```lua
function f(a, b)
    print(a, b)
end
f() -- prints nil, nil
f(1) -- prints 1, nil
f(1, 2) -- prints 1, 2
f(1, 2, 3) -- prints 1, 2
```

Reply:
OK.